From c476301448fc5688296130c734e3c552b7b6be2b Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 23 Nov 2023 23:36:21 +0100
Subject: [PATCH 0081/1695] pta: stats: initialize local variables where
 defined

Initialize local variables where defined in the statistics PTA.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1851a010e21a409059a4002f23c0c2f8f1559608 pta: stats: initialize local variables where defined]
Change-Id: I7e84cbe3a895bced0d499dd1d7856fa61b5a61aa
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 12f658b9e912d3f5934a07291e48ab2895712cd0]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/pta/stats.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/core/pta/stats.c b/core/pta/stats.c
index 9abcd981c..e0e8b9083 100644
--- a/core/pta/stats.c
+++ b/core/pta/stats.c
@@ -53,10 +53,10 @@
 
 static TEE_Result get_alloc_stats(uint32_t type, TEE_Param p[TEE_NUM_PARAMS])
 {
-	struct pta_stats_alloc *stats;
-	uint32_t size_to_retrieve;
-	uint32_t pool_id;
-	uint32_t i;
+	struct pta_stats_alloc *stats = NULL;
+	uint32_t size_to_retrieve = 0;
+	uint32_t pool_id = 0;
+	uint32_t i = 0;
 
 	/*
 	 * p[0].value.a = pool id (from 0 to n)
@@ -130,7 +130,7 @@ static TEE_Result get_alloc_stats(uint32_t type, TEE_Param p[TEE_NUM_PARAMS])
 
 static TEE_Result get_pager_stats(uint32_t type, TEE_Param p[TEE_NUM_PARAMS])
 {
-	struct tee_pager_stats stats;
+	struct tee_pager_stats stats = { };
 
 	if (TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_OUTPUT,
 			    TEE_PARAM_TYPE_VALUE_OUTPUT,
-- 
2.47.2

